app/src/ingest_bem_pdfs.py

```diff
@@ -67,13 +68,20 @@ def _ingest_bem_pdfs(
             _add_embeddings(chunks)
             db_session.add_all(chunks)
 
-            _save_json(file_path, chunks)
+            if save_json:
+                # Note that chunks are being added to the DB before saving the JSON.
+                # Originally, we thought about reviewing the JSON manually before adding chunks to the DB.
+                _save_json(file_path, chunks)
 
 
-def _parse_pdf(file: BinaryIO) -> list[EnrichedText]:
+def _parse_pdf(file: BinaryIO, file_path: str) -> list[EnrichedText]:
```

Comment:
I'd slightly prefer not to pass in a parameter if it's only being used in a log message -- sequentially this log message follows `logger.info("Processing file: %s", file_path)` to help with tracing (though I agree logging file_path directly in `Failed to extract and associate stylings` is nicer)

Reply:
I concur. Given the exploratory nature of the BEM PDFs, I'll keep this in. For future implementations, passing around a single ParsingContext that includes the filename, current headings, page number, etc. would be helpful.